src/Microsoft.AspNetCore.StaticFiles/Infrastructure/SharedOptionsBase.cs

```diff
@@ -11,8 +10,7 @@ namespace Microsoft.AspNetCore.StaticFiles.Infrastructure
     /// <summary>
     /// Options common to several middleware components
     /// </summary>
-    /// <typeparam name="T">The type of the subclass</typeparam>
-    public abstract class SharedOptionsBase<T>
+    public abstract class SharedOptionsBase
```

Comment:
It was when there were chainable set methods, the T on the base was used as the return type so chaining didn't downcast.